docs/tutorials/custom_experiment.rst

```diff
@@ -0,0 +1,362 @@
+
+Our objective is to create a new class, ``HigherOrderFineXAmplitude``, which calibrates 
+schedules on transitions other than the 0 <-> 1 transition for the :math:`\pi` rotation.
+In order to do this, we need to create a subclass as shown below.
+
+.. code-block::
+   
+    class HigherOrderFineXAmplitude(FineXAmplitude):
+        def _pre_circuit(self) -> QuantumCircuit:
+            """Return a preparation circuit.
+            
+            This method can be overridden by subclasses e.g. to calibrate schedules on
+            transitions other than the 0 <-> 1 transition.
+            """
+            circuit = QuantumCircuit(1)
+
+            circuit.x(0)
+
+            if self.experiment_options.add_sx:
+                circuit.sx(0)
+
+            if self.experiment_options.sx_schedule is not None:
+                sx_schedule = self.experiment_options.sx_schedule
+                circuit.add_calibration("sx", (self.physical_qubits[0],), sx_schedule, params=[])
+                circuit.barrier()
+
+            return circuit
+
+In this subclass we have overridden the ``_pre_circuit`` method in order to calibrate on higher energy transitions by using an initial X gate to populate the first excited state.
+
+Using the Subclass
+------------------
+
+Now, we can use our new subclass as we would the original parent class.
+Shown below are results from following the :ref:`fine-amplitude-cal` tutorial
+for detecting an over-rotated pulse using our new 
+``HigherOrderFineXAmplitude`` class in place of the original 
+:class:`.FineXAmplitude` class.
+You can try this for yourself and verify that your results are similar.
+
+.. code-block::
+   
+   DbAnalysisResultV1
+   - name: d_theta
+   - value: -0.020710672666906425 ± 0.0012903658449026907
+   - χ²: 0.7819653845899581
+   - quality: good
+   - device_components: ['Q0']
+   - verified: False
+
+Writing a custom experiment
+---------------------------
+
+Now we'll use what we've learned so far to make a full custom experiment from
+the :class:`.BaseExperiment` template.
+
+A randomized measurement experiment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This experiment creates a list of copies of an input circuit
+and randomly samples an N-qubit Pauli to apply to each one before
+a final N-qubit Z-basis measurement to randomize the expected
+ideal output bitstring in the measurement.
+
+The analysis uses the applied Pauli frame of a randomized
+measurement experiment to de-randomize the measured counts
+and combine across samples to return a single counts dictionary
+the original circuit. This has the effect of Pauli-twirling and symmetrizing the
+measurement readout error. 
+
+.. jupyter-execute::
+
+  from numpy.random import default_rng, Generator
+  from qiskit import QuantumCircuit
+  from qiskit_experiments.framework import BaseExperiment
+  from qiskit.quantum_info import random_pauli_list
```

Comment:
Consider arranging the imports, like pylint usually requests. Do we have pylint for the code snippets?

Reply:
Fixed these. I guess we can run a linter on the generated jupyter notebooks, but I don't know if we want to add that to the CI.